src/eigen.jl

```diff
@@ -13,6 +13,21 @@ end
     end
 end
 
+function eigvals(A::StaticMatrixLike, B::StaticMatrixLike; kwargs...)
+    SA = Size(A)
+    if SA != Size(B)
+        error("Generalized eigenvalues can only be calculated for matrices of equal sizes")
+    end
+    if SA[1] != SA[2]
+        error("Generalized eigenvalues can only be calculated for square matrices")
```

Comment:
There's also `checksquare` for this.  Though the error message you've got here is a bit more specific (should be a `DimensionMismatch`).

(See also the message generated by `LinearAlgebra` for these cases.)